py/desiutil/bitmask.py

```diff
@@ -0,0 +1,162 @@
+"""
+desiutil.maskbits
+=================
+
+Mask bits for the spectro pipeline.
+
+Individual packages will define their own mask bits and use this as a utility
+access wrapper.  Typical users will not need to construct BitMask objects on
+their own.
+
+Stephen Bailey, Lawrence Berkeley National Lab
+Fall 2015
+
+Example::
+
+    #- Creating a BitMask
+    from desiutil.bitmask import BitMask
+
+    import yaml
+    _bitdefs = yaml.load('''
+    ccdmask:
+        - [BAD,       0, "Pre-determined bad pixel (any reason)"]
+        - [HOT,       1, "Hot pixel"]
+        - [DEAD,      2, "Dead pixel"]
+        - [SATURATED, 3, "Saturated pixel from object"]
+        - [COSMIC,    4, "Cosmic ray"]
+    ''')
+    ccdmask = BitMask('ccdmask', _bitdefs)
+
+    #- Accessing the mask
+    ccdmask.COSMIC | ccdmask.SATURATED  #- 2**4 + 2**3
+    ccdmask.mask('COSMIC')     #- 2**4, same as ccdmask.COSMIC
+    ccdmask.mask(4)            #- 2**4, same as ccdmask.COSMIC
+    ccdmask.COSMIC             #- 2**4, same as ccdmask.mask('COSMIC')
+    ccdmask.bitnum('COSMIC')   #- 4
+    ccdmask.bitname(4)         #- 'COSMIC'
+    ccdmask.names()            #- ['BAD', 'HOT', 'DEAD', 'SATURATED', 'COSMIC']
+    ccdmask.names(3)           #- ['BAD', 'HOT']
+    ccdmask.comment(0)         #- "Pre-determined bad pixel (any reason)"
+    ccdmask.comment('COSMIC')  #- "Cosmic ray"
+"""
```

Comment:
This example might make a good [doctest](https://docs.python.org/2/library/doctest.html).